### DISCUSSION TITLE
Check if music file is broken or not

### DISCUSSION THREAD

**Discussion by qarmin** [pdeljanov/Symphonia#121] [General]
Basing on code from https://github.com/pdeljanov/Symphonia/blob/master/GETTING_STARTED.md#decode-loop I created my own variation to check if audio file is broken or not.

Looks that this implementation have some problems:
- At the end I use "UnexpectedEof" to check for Eof, but as name suggest this should be unexpected. How can I check for valid/expected eof in proper way?
- If symphonia doesn't support valid audio file(ffprobe doesn't show anything bad) due returns "probe info not available", then should I report bug? - example file - [ 51 - Beorunna's Well.mp3.zip](https://github.com/pdeljanov/Symphonia/files/8643182/51.-.Beorunna.s.Well.mp3.zip)

```
fn parse_audio_file(file_handler: File) -> std::result::Result<(), Error> {
    let mss = MediaSourceStream::new(Box::new(file_handler), Default::default());

    let probed = match symphonia::default::get_probe().format(
        &Default::default(),
        mss,
        &Default::default(),
        &Default::default(),
    ) {
        Ok(t) => t,
        Err(_) => return Err(Error::Unsupported("probe info not available")),
    };

    let mut format = probed.format;

    let track = match format
        .tracks()
        .iter()
        .find(|t| t.codec_params.codec != CODEC_TYPE_NULL)
    {
        Some(k) => k,
        None => return Err(Error::Unsupported("not supported audio track")),
    };

    let mut decoder =
        match symphonia::default::get_codecs().make(&track.codec_params, &Default::default()) {
            Ok(k) => k,
            Err(_) => return Err(Error::Unsupported("not supported codec")),
        };

    loop {
        let packet = match format.next_packet() {
            Ok(packet) => packet,
            Err(Error::ResetRequired) => {
                return Err(Error::ResetRequired);
            }
            Err(err) => {
                if let IoError(ref er) = err {
                    // Catch eof, not sure how to do it properly
                    if er.kind() == io::ErrorKind::UnexpectedEof {
                        return Ok(());
                    }
                }
                return Err(err);
            }
        };

        decoder.decode(&packet)?;
    }
}
```

**Comment by pdeljanov** [comment]
Hi @qarmin,

Regarding the issues you are having:

1. Symphonia returns `UnexpectedEof` whether the ending was expected or unexpected. For many media formats (for example, MP3, some AACs), there is not any distinction between the two since they don't tell us their intended length.
2. If you find files that are broken for any of the supported formats or codecs, please report them. I can confirm the one you attached is indeed not working. I will try to fix it.
